main/espFeatures/motorFeature.h

```diff
@@ -254,10 +261,33 @@ class MotorFeature : public Next {
     using MotorClass = jac::Class<MotorProtoBuilder<MotorFeature<Next>>>;
 #endif
 
+
+
     MotorFeature() {
         MotorClass::init("Motor");
     }
 
+    ~MotorFeature() {
+        for(auto& motor : _motors) {
+            auto& jsdcMotor = *MotorProtoBuilder<MotorFeature<Next>>::getOpaque(this->context(), motor);
+            jsdcMotor.clear();
+        }
+    }
```

Comment:
If the `pendingPromise` is not freed BEFORE `JS_FreeRuntime`, quickjs goes into assert thinking there is a memory leak. It would actually free the promises later, but during the leak check, references from imported modules are still alive, so the leak check happens before the `export const` reference on motor is freed and before `JSDCMotor` finalizer is called.

So it's the leak check being wrong, but it means it crashes program during uploading if you use motors but don't resolve the promise.